### ISSUE TITLE
Segfault when splitting coastline

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#2901]
**[Submitted to the original trac issue database at 8.00pm, Tuesday, 20th April 2010]**

When I split way_3492478 at any node I get a segfault:

```
#0  0x0000000000000000 in ?? ()
#1  0x000000000055edbe in Layer::getFeatureSet (this=<value optimized out>, theFeatures=..., theCoastlines=<value optimized out>, 
    invalidRects=<value optimized out>, clipRect=<value optimized out>, theProjection=..., theTransform=...) at Layers/Layer.cpp:109
#2  0x00000000004b470d in MapView::buildFeatureSet (this=0xab37d0) at MapView.cpp:320
#3  0x00000000004bcec4 in MapView::paintEvent (this=0xab37d0, anEvent=0x7fffffffa720) at MapView.cpp:198
#4  0x00007ffff4b82822 in QWidget::event(QEvent*) () from /usr/lib/libQtGui.so.4
#5  0x00000000004b66f1 in MapView::event (this=0xab37d0, event=0x7fffffffa720) at MapView.cpp:950
#6  0x00007ffff4b2c8fc in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib/libQtGui.so.4
#7  0x00007ffff4b32ddb in QApplication::notify(QObject*, QEvent*) () from /usr/lib/libQtGui.so.4
#8  0x00007ffff42ebf1c in QCoreApplication::notifyInternal(QObject*, QEvent*) () from /usr/lib/libQtCore.so.4
#9  0x00007ffff4b8ae9d in QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) ()
   from /usr/lib/libQtGui.so.4
#10 0x00007ffff4b8bb38 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) () from /usr/lib/libQtGui.so.4
#11 0x00007ffff4b8b969 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) () from /usr/lib/libQtGui.so.4
#12 0x00007ffff4b8b969 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) () from /usr/lib/libQtGui.so.4
#13 0x00007ffff4b8b969 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) () from /usr/lib/libQtGui.so.4
#14 0x00007ffff4b8b969 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) () from /usr/lib/libQtGui.so.4
#15 0x00007ffff4b8b969 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) () from /usr/lib/libQtGui.so.4
#16 0x00007ffff4b8b969 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice*, QList<QObject*> const&, int, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) () from /usr/lib/libQtGui.so.4
#17 0x00007ffff4b8abfa in QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) ()
   from /usr/lib/libQtGui.so.4
#18 0x00007ffff4d445a5 in ?? () from /usr/lib/libQtGui.so.4
#19 0x00007ffff4b7c210 in QWidgetPrivate::syncBackingStore() () from /usr/lib/libQtGui.so.4
#20 0x00007ffff4b82f25 in QWidget::event(QEvent*) () from /usr/lib/libQtGui.so.4
#21 0x00007ffff4f43f6b in QMainWindow::event(QEvent*) () from /usr/lib/libQtGui.so.4
#22 0x00007ffff4b2c8fc in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib/libQtGui.so.4
#23 0x00007ffff4b32ddb in QApplication::notify(QObject*, QEvent*) () from /usr/lib/libQtGui.so.4
#24 0x00007ffff42ebf1c in QCoreApplication::notifyInternal(QObject*, QEvent*) () from /usr/lib/libQtCore.so.4
#25 0x00007ffff4d418f7 in ?? () from /usr/lib/libQtGui.so.4
#26 0x00007ffff4d44da6 in ?? () from /usr/lib/libQtGui.so.4
#27 0x00007ffff4b7eb3b in QWidget::repaint(QRect const&) () from /usr/lib/libQtGui.so.4
#28 0x00007ffff4b7ec03 in QWidget::repaint() () from /usr/lib/libQtGui.so.4
#29 0x00007ffff4ee1918 in ?? () from /usr/lib/libQtGui.so.4
#30 0x00007ffff4ee1bac in QAbstractButton::mouseReleaseEvent(QMouseEvent*) () from /usr/lib/libQtGui.so.4
#31 0x00007ffff4fadc8a in QToolButton::mouseReleaseEvent(QMouseEvent*) () from /usr/lib/libQtGui.so.4
#32 0x00007ffff4b82c42 in QWidget::event(QEvent*) () from /usr/lib/libQtGui.so.4
#33 0x00007ffff4b2c8fc in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib/libQtGui.so.4
#34 0x00007ffff4b335ab in QApplication::notify(QObject*, QEvent*) () from /usr/lib/libQtGui.so.4
#35 0x00007ffff42ebf1c in QCoreApplication::notifyInternal(QObject*, QEvent*) () from /usr/lib/libQtCore.so.4
#36 0x00007ffff4b3278e in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool) () from /usr/lib/libQtGui.so.4
#37 0x00007ffff4bb23cb in ?? () from /usr/lib/libQtGui.so.4
#38 0x00007ffff4bb0d3c in QApplication::x11ProcessEvent(_XEvent*) () from /usr/lib/libQtGui.so.4
#39 0x00007ffff4bdc7e2 in ?? () from /usr/lib/libQtGui.so.4
#40 0x00007fffee9646c2 in g_main_context_dispatch () from /lib/libglib-2.0.so.0
#41 0x00007fffee968538 in ?? () from /lib/libglib-2.0.so.0
#42 0x00007fffee9686ec in g_main_context_iteration () from /lib/libglib-2.0.so.0
#43 0x00007ffff4315373 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
#44 0x00007ffff4bdc3de in ?? () from /usr/lib/libQtGui.so.4
#45 0x00007ffff42ea842 in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
#46 0x00007ffff42eac1c in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
#47 0x00007ffff42ee95b in QCoreApplication::exec() () from /usr/lib/libQtCore.so.4
#48 0x00000000004defb7 in main (argc=1, argv=<value optimized out>) at Main.cpp:222
```

I don't know whether it's significant that it belongs to a relation - deleting the relation doesn't seem to help, though.  Other ways split fine, including other bits of coastline.

**Comment by tomhughes** [comment]
**Author: TomH**
**[Added to the original trac issue at 8.07pm, Tuesday, 20th April 2010]**

I assume this refers to merkaartor...

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 8.34am, Wednesday, 21st April 2010]**

(In [20950]) FIX : splitting roads part of a relation (fixes #2901)
